postgresql.go

```diff
@@ -442,34 +442,42 @@ func (h *postgreSQL) buildInsertSQL(q shared.Queryable, tableName string, column
 	return h.baseHelper.buildInsertSQL(q, tableName, columns, values)
 }
 
-func (h *postgreSQL) tableHasIdentityColumn(q shared.Queryable, tableName string) (bool, error) {
+func (h *postgreSQL) tableHasIdentityColumn(tableName string) bool {
+	return h.tablesHasIdentityColumn[tableName]
+}
+
+func (h *postgreSQL) buildTableHasIdentityColumn(q shared.Queryable) error {
 	defer h.tablesHasIdentityColumnMutex.Unlock()
 	h.tablesHasIdentityColumnMutex.Lock()
 
-	hasIdentityColumn, exists := h.tablesHasIdentityColumn[tableName]
-	if exists {
-		return hasIdentityColumn, nil
-	}
+	const query = `SELECT table_name, COUNT(*) AS identity_column_count
+    FROM information_schema.columns
+    WHERE
+      table_schema NOT IN ('pg_catalog', 'information_schema', 'crdb_internal') AND
```

Comment:
is those new checks necessary?

Reply:
Not necessarily, but we don't need to store that information, don't we?

Reply:
Yes, true